### PR TITLE
#1798 - Dependency Updates

### PR DESCRIPTION
**Pull request by k-macmillan**
# Description
Ran `poetry update` and validated all tests passed locally.

```
Updating dependencies
Resolving dependencies... (12.4s)

Package operations: 0 installs, 39 updates, 0 removals

  • Updating urllib3 (2.2.1 -> 2.2.2)
  • Updating botocore (1.34.78 -> 1.34.127)
  • Updating idna (3.6 -> 3.7)
  • Updating blinker (1.7.0 -> 1.8.2)
  • Updating itsdangerous (2.1.2 -> 2.2.0)
  • Updating jinja2 (3.1.3 -> 3.1.4)
  • Updating rpds-py (0.18.0 -> 0.18.1)
  • Updating werkzeug (3.0.2 -> 3.0.3)
  • Updating zipp (3.18.1 -> 3.19.2)
  • Updating aiohttp (3.9.3 -> 3.9.5)
  • Updating boto3 (1.34.78 -> 1.34.127)
  • Updating certifi (2024.2.2 -> 2024.6.2)
  • Updating exceptiongroup (1.2.0 -> 1.2.1)
  • Updating flask (3.0.2 -> 3.0.3)
  • Updating importlib-metadata (7.0.0 -> 7.1.0)
  • Updating mako (1.3.2 -> 1.3.5)
  • Updating prompt-toolkit (3.0.43 -> 3.0.47)
  • Updating redis (5.0.3 -> 5.0.6)
  • Updating referencing (0.34.0 -> 0.35.1)
  • Updating typing-extensions (4.11.0 -> 4.12.2)
  • Updating bcrypt (4.1.2 -> 4.1.3)
  • Updating cryptography (42.0.5 -> 42.0.8)
  • Updating kombu (5.3.6 -> 5.3.7)
  • Updating opentelemetry-api (1.24.0 -> 1.25.0)
  • Updating packaging (24.0 -> 24.1)
  • Updating parse (1.20.1 -> 1.20.2)
  • Updating pypdf (4.1.0 -> 4.2.0)
  • Updating requests (2.31.0 -> 2.32.3)
  • Updating alembic-utils (0.8.2 -> 0.8.3)
  • Updating authlib (1.3.0 -> 1.3.1)
  • Updating cachelib (0.12.0 -> 0.13.0)
  • Updating celery (5.3.6 -> 5.4.0)
  • Updating flask-cors (4.0.0 -> 4.0.1)
  • Updating gunicorn (21.2.0 -> 22.0.0)
  • Updating jsonschema (4.21.1 -> 4.22.0)
  • Updating notification-utils (2.0.1 6044890 -> 2.0.1 20140c9)
  • Updating notifications-python-client (9.0.0 -> 9.1.0)
  • Updating sentry-sdk (1.44.1 -> 2.5.1)
  • Updating twilio (9.0.4 -> 9.1.1)

Writing lock file
```

issue #1798 

## Type of change

Please check the relevant option(s).
- [x] Dependency updates
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

Local testing:
```
# Parallel
=========== 2795 passed, 58 skipped, 17 xfailed in 109.54s (0:01:49) ===========
# Serial
=== 157 passed, 3 skipped, 2870 deselected, 13 xfailed in 120.23s (0:02:00) ====

```
image

[Deployed](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/9552641218), passed all regression and unit tests.

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [ ] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] The ticket is now moved into the DEV test column

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,11 +32,10 @@ install-safety:
 
 check-dependencies: install-safety ## Scan dependencies for security vulnerabilities
 	# 12 Dec 2023: 51668 is fixed with >= 2.0.0b1 of SQLAlchemy. Ongoing refactor to upgrade.
-	# 6 June 2024: 70624 will be resolved with ticket #1794
+	# 6 June 2024: 70624 will be resolved with ticket #1794 - flask-cors
```
Comment (cris-oddball) [inline Makefile:35]: flask-cors was updated in this PR to 4.0.1, which clears the vulnerability. l35 can be deleted.

Comment (cris-oddball) [inline Makefile:35]: flask-cors was updated in this PR to 4.0.1, which clears the vulnerability. l35 can be deleted.

File: Makefile [hunk 2]
```diff
@@ -32,11 +32,10 @@ install-safety:
...
 	# 6 June 2024: 70612 vulnerability found with jinja2 version 3.1.3
-	# 14 June 2024: 70813 Vulnerability found in flask-cors version 4.0.0
 
-	safety check -r poetry.lock --full-report -i 51668,70624,70612,70813
+	safety check -r poetry.lock --full-report -i 51668,70612,70624
```
Comment (cris-oddball) [inline Makefile:38]: in line 38, 70624 and 70624 can be removed, leaving just sqlalchemy.

**Review by cris-oddball — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (k-macmillan) [reply Makefile:35]: [I wish](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/9552570711/job/26329453696#step:4:212).

I deleted all of them and tested, it still flagged all three that are currently there.

Comment (cris-oddball) [reply Makefile:35]: MEH

**Review by cris-oddball — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Pull request merged**